data/tabular/train_test_split.py

```diff
@@ -1,209 +1,459 @@
+                return "valid"
+
+    # we err toward doing more I/O but having simpler code to ensure we don't make anything stupid
+    all_yaml_files = get_yaml_files(data_dir)
+    smiles_yaml_files = [
+        file for file in all_yaml_files if has_yaml_file_smiles_column(file)
+    ]
+    # we filter those out that are in the to_scaffold_split list
+    not_scaffold_split_yaml_files = [
+        file for file in smiles_yaml_files if not is_in_scaffold_split_list(file)
+    ]
+
+    # if we debug, we only run split on the first 5 datasets
+    if debug:
+        not_scaffold_split_yaml_files = not_scaffold_split_yaml_files[:5]
+
+    # we run random splitting for each dataset but ensure that the validation and test sets
+    # do not contain any SMILES that are in the validation and test sets of the scaffold split
+
+    with open(val_smiles_path, "r") as f:
+        val_smiles = f.read().splitlines()
+
+    with open(test_smiles_path, "r") as f:
+        test_smiles = f.read().splitlines()
+
+    # some datasets are hundreds of GB, so we have to use dask
+    # we will first do a random split on the whole dataset
+    # then we flip the values based on whether a SMILES
+    # is in the validation or test set of the scaffold split
+    # we will then write the new data_clean.csv file
+
+    # if the is no data_clean.csv file, we run the transform.py script
+    # in the directory of the yaml file
+    # we will then read the data_clean.csv file and do the split
+    for file in not_scaffold_split_yaml_files:
+        print(f"Processing {file}")
+        run_transform(file)
+        ddf = dask.dataframe.read_csv(
+            os.path.join(os.path.dirname(file), "data_clean.csv")
         )
-    else:
-        split_ = pd.DataFrame(
-            np.random.choice(
-                ["train", "test", "valid"],
-                size=len(repr_df),
-                p=[1 - val_size - test_size, test_size, val_size],
+        meta = ("split", "object")  # Meta defines the structure of the new column
+        ddf["split"] = ddf.apply(
+            assign_split,
+            axis=1,
+            meta=meta,
+            args=(test_smiles, val_smiles, train_frac, test_frac, dask_random_state),
+        )
+        split_counts = ddf["split"].value_counts().compute()
+
+        print(
+            f"Dataset {file} has {len(ddf)} datapoints. Split sizes: {split_counts['train']} train, {split_counts['valid']} valid, {split_counts['test']} test."  # noqa: E501
+        )
+
+        # we then write the new data_clean.csv file
+        # if the override option is true, we write this new file to `data_clean.csv` in the same directory
+        # otherwise, we copy the old `data_clean.csv` to `data_clean_old.csv`
+        # and write the new file to `data_clean.csv`
+        if not override:
+            os.rename(
+                os.path.join(os.path.dirname(file), "data_clean.csv"),
+                os.path.join(os.path.dirname(file), "data_clean_old.csv"),
```

Comment:
same as above: maybe add another arg to not keep the old file, as I guess we will run into storage issues with that

Reply:
if you override it overrides the old file?